lib/models/os.coffee

```diff
@@ -35,29 +41,27 @@ OSParams = require('./os-params')
 # parameters =
 #		network: 'ethernet'
 #		appId: 91
+#		onProgress: (state) ->
+#			return if not state?
+#			console.log "Total: #{state.total}"
+#			console.log "Received: #{state.received}"
 #
 # resin.models.os.download parameters, '/opt/os.zip', (error) ->
 #		throw error if error?
-#	, (state) ->
-#		return if not state?
-#		console.log "Total: #{state.total}"
-#		console.log "Received: #{state.received}"
 ###
-exports.download = (parameters, destination, callback, onProgress) ->
+exports.download = (options, destination, callback) ->
+	Promise.try ->
+		if not token.getUsername()?
+			throw new errors.ResinNotLoggedIn()
 
-	if not token.getUsername()?
-		return callback(new errors.ResinNotLoggedIn())
+		osParams = _.omit(options, 'onProgress')
```

Comment:
@Page- `os-params` already whitelists the options and throws an error if you're passing an extra one: https://github.com/resin-io/resin-sdk/blob/master/lib/models/os-params.coffee#L62. Therefore the `omit` here is to prevent an error being thrown for `onProgress`.